src/modules/auth/hooks/queries/useGetInitializationFees.js

```diff
@@ -0,0 +1,30 @@
+import { useInvokeQuery } from '@common/hooks';
+import { useTokensBalance } from 'src/modules/token/fungible/hooks/queries';
+import { useAuth } from './useAuth';
+
+export const useGetInitializationFees = ({ options = {}, address } = {}) => {
+  const config = {
+    data: {
+      endpoint: 'token_getInitializationFees',
+      params: {},
+    },
+  };
+
+  const queryConfig = {
+    options: { enabled: !!address },
+    config: { params: { address } },
+  };
+
+  const { data: auth, isLoading: isAuthLoading } = useAuth(queryConfig);
+  const { data: token, isLoading: isTokenLoading } = useTokensBalance(queryConfig);
+
+  const isAccountInitialized = auth?.data?.nonce !== '0' && +token?.data?.[0]?.availableBalance;
+  const shouldReturnInitalizationFee = !isAccountInitialized && !isAuthLoading && !isTokenLoading;
```

Comment:
and its also used to determine if we should return a fee value from the query's catch because react-query would still return a cached value even though the `enable` flag on the query's option has been set to `false`